frontend/src/context/AuthContext.jsx

```diff
@@ -16,17 +16,17 @@ const AuthContext = createContext();
 export const AuthProvider = ({ children }) => {
     const [currentUser, setCurrentUser] = useState(null);
     const [loading, setLoading] = useState(true);
-}
+
 
 // listens for auth state change 
-useEffect(() => {
-    const unsubscribe = onAuthStateChanged(auth, user => {
-        setCurrentUser(user);
-        setLoading(false);
-    });
+    useEffect(() => {
+        const unsubscribe = onAuthStateChanged(auth, user => {
+            setCurrentUser(user);
+            setLoading(false);
+        });
 
-    return () => unsubscribe();
-}, []);
+        return () => unsubscribe();
+    }, []);
 
 // auth functions
 const register = (email, password) => createUserWithEmailAndPassword(auth, email,password);
```

Comment:
seems we're in the risk of overloading Context here, please make sure we only include the props that needs to be passed deeply in the Context, in this case, `currentUser` is one example. Please try moving all the other functions out of the context.